### PR TITLE
chore: add dev container

### PR DESCRIPTION
**Pull request by ctran88**
## What's New?

- Adds a dev container config file to bootstrap the dev environment
- Adds rubocop to dev dependencies to satisfy Ruby LSP VS Code extension error when setting rubocop as its formatter
- Replaces openapitools npm package with docker image to codegen because there were more issues than I had time to resolve with putting ruby and node/npm in the same container

**To test:**
1. In VS Code, download the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
2. If a popup on your lower right window shows up that asks to **"Reopen in Container"**, select it. Skip to step 5
    
    image

3. `cmd + shift + P` to bring up command palette
4. Start typing "rebuild" and select **"Dev Containers: Rebuild and Reopen in Container"**
5. The first time may take a minute or two as it downloads and builds the docker image. Progress can be tracked by clicking the popup

    image

6. Make sure you have your `.env` file populated 
7. Try running linter, formatter, and tests by following commands in the `on-pull-request.yml` CI file or following the documentation for the tools being used
8. Check the extensions and you should find the basics of what you need to dev

**Without VS Code**
If you want to try it without VS Code, you can do so by using the [Dev Container CLI](https://github.com/devcontainers/cli?tab=readme-ov-file#npm-install)

**Codegen change**
codegen change can be tested with
```
curl -o openapi.json https://api.swaggerhub.com/apis/passage/passage-mgmt-sdk-api/1
./generate.sh openapi.json
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have manually tested my code thoroughly
- [ ] I have added/updated inline documentation for public facing interfaces if relevant
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing integration and unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+	"name": "Ruby",
+	"image": "mcr.microsoft.com/devcontainers/ruby:1-3.3-bullseye",
+	"features": {
+		"ghcr.io/itsmechlark/features/act:1": {},
+		"ghcr.io/devcontainers/features/docker-in-docker:2": {}
```
Comment (ctran88) [inline .devcontainer/devcontainer.json:6]: this is to enable the openapitools docker image usage

**Comment by ctran88** [comment]
linting and formatting is expected to fail since that is addressed in #101

File: openapitools.json [hunk 2]
```diff
@@ -1,7 +0,0 @@
-{
-  "$schema": "./node_modules/@openapitools/openapi-generator-cli/config.schema.json",
-  "spaces": 2,
-  "generator-cli": {
-    "version": "7.1.0"
-  }
-}
```
Comment (vanessa-passage) [inline openapitools.json:7]: Is it okay to remove this file? Are there any side effects?

Comment (ctran88) [reply openapitools.json:7]: No side effects that I saw when I ran the generator

**Review by vanessa-passage — APPROVED** [review]
👍  Tested

**Pull request merged**